ci/common_install.sh

```diff
@@ -51,7 +51,8 @@ if [ ! -d boost-root ]; then
 else
     cd boost-root
     git checkout $BOOST_BRANCH
-    git pull
+    git pull --no-recurse-submodules
+    git submodule update
```

Comment:
I feel that this is superflous: We pull/checkout the submodules via `depinst.py` or `git submodule update -q --init tools/boostdep` below already so this might rather increase the runtime
Although I'm not sure if `depinst.py` does an `submodule update` which would be required in this case.

Reply:
This is the drone caching code path, which is rarely followed. It would only be hit if you have enabled caching.  
Then, it's replacing a "git clone".  Since "git clone" downloads a fresh up-to-date copy of the repo, it's mimicking that action with a "git pull", to update the local files. Maybe the submodule part here is superfluous. not sure. A standard "git pull" does in fact update submodules you have checked out.   It goes fast if no action is needed.